src/cogent3/core/new_alignment.py

```diff
@@ -2374,3 +2379,559 @@ def _(
+        *,
+        seqid: str,
+        start: OptInt = None,
+        stop: OptInt = None,
+    ) -> str:
+        """Return sequence corresponding to seqid as a string.
+        start/stop are in alignment coordinates. Includes gaps."""
+
+        return self.alphabet.from_indices(
+            self.get_gapped_seq_array(seqid=seqid, start=start, stop=stop)
+        )
+
+    def get_seq_bytes(
+        self,
+        *,
+        seqid: str,
+        start: OptInt = None,
+        stop: OptInt = None,
+    ) -> bytes:
+        """Return sequence corresponding to seqid as a bytes string.
+        start/stop are in alignment coordinates. Excludes gaps."""
+        return self.get_seq_str(seqid=seqid, start=start, stop=stop).encode("utf8")
+
+    def get_gapped_seq_bytes(
+        self,
+        *,
+        seqid: str,
+        start: OptInt = None,
+        stop: OptInt = None,
+    ) -> bytes:
+        """Return sequence corresponding to seqid as a bytes string.
+        start/stop are in alignment coordinates. Includes gaps."""
+        return self.get_gapped_seq_str(seqid=seqid, start=start, stop=stop).encode(
+            "utf8"
+        )
+
+    def add_seqs(
+        self, seqs: dict[str, StrORArray], force_unique_keys=True
+    ) -> AlignedData:
+        """Returns a new AlignedData object with added sequences."""
+        if force_unique_keys and any(name in self.names for name in seqs):
+            raise ValueError("One or more sequence names already exist in collection")
+        new_data = {
+            **self.seqs,
+            **seqs,
+        }
+        return self.__class__.from_aligned_seqs(new_data, self.moltype)
+
+    def reversed(self):
+        # todo: kath
+        ...
+
+    def subset(self):
+        # todo: kath
+        ...
+
+    def to_alphabet(self):
+        # todo: kath
+        ...
+
+    def to_rich_dict(self):
+        # todo: kath
+        ...
+
+
+class AlignedDataView(new_sequence.SeqViewABC, new_sequence.SliceRecordABC):
+    # refactor: docstring
+    def __init__(
+        self,
+        *,
+        seq: AlignedData,
+        seqid: str,
+        seq_len: int,
+        start: OptInt = None,
+        stop: OptInt = None,
+        step: OptInt = None,
+        offset: int = 0,
+    ):
+        if step and step < 1:
+            raise ValueError(f"step cannot be {step}")
+        step = 1 if step is None else step
+        self.seq = seq
+        self._seq_len = self._checked_seq_len(seq_len)
+        func = (
+            new_sequence._input_vals_pos_step
+            if step > 0
+            else new_sequence._input_vals_neg_step
+        )
+        start, stop, step = func(self._seq_len, start, stop, step)
+        self.start = start
+        self.stop = stop
+        self.step = step
+        self._offset = offset
+        self._seqid = seqid
+
+    @property
+    def seqid(self) -> str:
+        return self._seqid
+
+    @property
+    def seq_len(self) -> int:
+        return self._seq_len
```

Comment:
Are we distinguishing between aligned length and ungapped sequence length here?

Reply:
I've renamed to `parent_len` - its the length of the aligned